### PR TITLE
unit test for login, signup function also cleared up the functions

### PR DESCRIPTION
**Pull request by ahmetfirat23**
I have written unit tests for login and signup. I only covered cases that are not handled in frontend. Also made some clearing in the functions too.  Please make sure to understand the changes in the functions before accepting. Also please work from here on authentication.

### CODE REVIEW

File: backend/main/views.py
Comment (isilsukarakuzuu) [inline backend/main/views.py:1]: Please don't remove the main-page endpoints from views.py.

**Review by ahmetfirat23 — COMMENTED** [review]
I think this is fine now

**Comment by ahmetfirat23** [comment]
I updated since I was asking to delete all previous work sry

**Review by isilsukarakuzuu — APPROVED** [review]
With new change, lgtm!

**Pull request merged**